pkg/controllers/dynakube/logmonitoring/reconciler.go

```diff
@@ -65,5 +68,31 @@ func (r *Reconciler) Reconcile(ctx context.Context) error {
 		return err
 	}
 
+	logMonitoringSettings, err := r.dtc.GetSettingsForLogModule(ctx, r.dk.Status.KubernetesClusterMEID, dtclient.LogMonitoringSettingsSchemaId)
+	if err != nil {
+		return errors.WithMessage(err, "error trying to check if setting exists")
+	}
+
+	if logMonitoringSettings.TotalCount > 0 {
+		log.Info("there are already settings", "settings", logMonitoringSettings)
+
+		conditions.SetLogMonitoringSettingExists(r.dk.Conditions(), conditionType)
+
+		return nil
+	} else if logMonitoringSettings.TotalCount == 0 {
+		matchers := []logmonitoring.IngestRuleMatchers{}
+		if len(r.dk.LogMonitoring().IngestRuleMatchers) > 0 {
+			matchers = r.dk.LogMonitoring().IngestRuleMatchers
+		}
+
+		objectId, err := r.dtc.CreateLogMonitoringSetting(ctx, dtclient.LogMonitoringSettingsSchemaId, r.dk.Status.KubernetesClusterMEID, r.dk.Status.KubernetesClusterName, matchers)
+
+		if err != nil {
+			return errors.WithMessage(err, "error when creating log monitoring setting")
+		}
+
+		log.Info("logmonitoring setting created", "settings", objectId)
+	}
+
```

Comment:
you should put this into it's own function

Reply:
done in https://github.com/Dynatrace/dynatrace-operator/pull/4053/commits/a4767e415761f112227f6ec964bd1ae7549845e4